### ISSUE TITLE
DOC: NEP copyright line is hard-code to 2017-2018 rather than <startdate> to <now>

### ISSUE THREAD

**Issue by eagunn** [numpy/numpy#26834]
### Issue with current documentation:

As I was proofing some other changes, I noticed that the Copyright line at the end of the published NEPS is hardcoded to 2017-2018.
![image](https://github.com/numpy/numpy/assets/6098961/8a115ada-a2ac-4b15-8e7c-a919eca21b89)

The main documentation more appropriately has a Copyright line that goes from 2008 to 2024.
![image](https://github.com/numpy/numpy/assets/6098961/29ef7e4c-94e6-4186-bfee-e2ba0bc70618)

This output is controlled by a copyright variable in the applicable conf.py file.

The NEPs' copyright variable in `doc/neps/conf.py1` is
`copyright = '2017-2018, NumPy Developers'`

I believe it should either be changed to be the same as the copyright variable in `doc/source/conf.py` which is:
`copyright = '2008-2024, NumPy Developers'`

OR, to simplify maintenance, it should be made dynamic using syntax something like:
```
import datetime
year = datetime.datetime.now().year
copyright = f'© 2008-{year}, NumPy Developers.'
```
(I haven't tested the above lines; they are adapted from ChatGPT code.)

I'd be happy to try making the dynamic date version of the change to the NEP conf.py since I'm working on those files now. It looks as if the appropriate start year for NEPs would be 2017. That's the current value and   appears to be the publication year of NEP 0.

**Comment by mattip** [comment]
What is the basis of adding any date whatsoever to the copyright notice? I could not find a solid reason to do this. Places like [this thread](https://stackoverflow.com/questions/2390230/do-copyright-dates-need-to-be-updated) state that a date is note needed, and even the copyright notice itself is informational: legally no statement is needed.

I would be in favor of removing the date, but would also support this idea of dynamically updating a range of dates `2008 - {year}, NumPy Developers` as a less controversial idea.

**Comment by eagunn** [comment]
@mattip It's a good question.

I had been unaware that, since 1989 (the date cited in your StackOverflow thread above and in [an informative article from the Copyright Alliance](https://copyrightalliance.org/education/copyright-law-explained/copyright-basics/what-is-copyright-notice/)), the whole copyright line is now considered optional. 

However, based on my experience as a content creator and sometime web dev, it is still a common convention to have one. I believe it provides potentially useful information to anyone who, via a link or a search, lands on a page without navigating there from the documentation home page. It's also conventional, in my experience, to use the format, on web pages, that asserts copyright through a range of years even though, technically, copyright applies only to individual instantiations of a work. 

I think it is important to eliminate the need for the conf.py value to be updated every year. I like the idea of using the dynamic date value that Python affords us but I don't feel strongly about that.

**Comment by rkern** [comment]
Whenever talking about whether something is "needed" or "optional", I find that it is always helpful to complete the sentence: "needed for what thing in particular". There are lots of things that a copyright notice can accomplish. It is true that under the Berne Convention, no copyright notice is needed for our copyright *to exist* on the work in Berne Convention countries; that exists as soon as the work is created. But the presence of a correctly-formatted copyright notice does have a legal effect in US courts. It removes the ability of the defendant to claim in court "I didn't know that it was validly copyrighted". By the letter of the [US Code](https://www.copyright.gov/title17/92chap4.html#401), this only applies to correctly-formatted copyright notices with a date, not dateless ones.

And for the record, because this is a common point of confusion reading this statute in the context of open source software, "the date of first publication" in a continually updated work is the date of the last substantive modification that you want to claim copyright on, not the unchanging first year. Each time we make substantive modifications, it's like making a new edition of a textbook, not just a new printing of a text unchanged from a decade ago. You include the date of publication of the new edition.

Since this is about the NEPs (sub-)website and not the NEPs themselves, the `copyright = f'© 2017-{year}, NumPy Developers.'` is probably sufficient, IMO.

**Comment by mattip** [comment]
It seems there is consensus that this would be a good idea
> it should be made dynamic using syntax something like:
```
import datetime
year = datetime.datetime.now().year
copyright = f'© 2008-{year}, NumPy Developers.'
```

so a PR would be welcome.

**Comment by eagunn** [comment]
@mattip Thank you for the nudge. I should be able to get a PR in for this today or tomorrow. 

Unless I hear otherwise, I'll make the changes to both conf.py files (one for the documentation and one for the NEP pages).

**Comment by eagunn** [comment]
Closing. Fixed by PR #26895.
